src/pages/fi-dim-values/index.js

```diff
@@ -0,0 +1,179 @@
+import { useContext, useEffect, useState } from 'react'
+import toast from 'react-hot-toast'
+import Table from 'src/components/Shared/Table'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useResourceQuery } from 'src/hooks/resource'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useWindow } from 'src/windows'
+import { ControlContext } from 'src/providers/ControlContext'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import DimValuesForm from './form/DimValuesForm'
+import { Grid } from '@mui/material'
+
+const DimensionsValues = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { stack } = useWindow()
+  const [tpaValues, setTpaValues] = useState([])
+  const [error, setError] = useState(false)
+
+  const [selectedTpaValue, setSelectedTpaValue] = useState({ key: null, value: '' })
+  const formatedRecordId = typeof selectedTpaValue?.key == 'string' ? selectedTpaValue.key.match(/\d+/)?.[0] : null
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+    refetch,
+    invalidate,
+    filterBy,
+    clearFilter,
+    access
+  } = useResourceQuery({
+    datasetId: ResourceIds.DimensionsValues,
+    filter: {
+      endpointId: FinancialRepository.DimensionValue.qry,
+      filterFn: fetchWithSearch
+    }
+  })
+
+  useEffect(() => {
+    if (formatedRecordId !== null) {
+      filterBy('qry', formatedRecordId)
+    } else {
+      clearFilter('qry')
+    }
+  }, [formatedRecordId])
+
+  async function fetchWithSearch({ filters }) {
+    if (formatedRecordId) {
+      const data = await getRequest({
+        extension: FinancialRepository.DimensionValue.qry,
+        parameters: `_filter=${filters.qry}&_dimension=${formatedRecordId}`
+      })
+
+      return data
+    }
+  }
+
+  const columns = [
+    {
+      field: 'id',
+      headerName: _labels.id,
+      flex: 1
+    },
+    {
+      field: 'name',
+      headerName: _labels.name,
+      flex: 1
+    }
+  ]
+
+  const del = async obj => {
+    try {
+      await postRequest({
+        extension: FinancialRepository.DimensionValue.del,
+        record: JSON.stringify(obj)
+      })
+      invalidate()
+      toast.success(platformLabels.Deleted)
+    } catch (error) {
+      toast.error(platformLabels.Error)
+    }
+  }
+
+  const add = () => {
+    if (formatedRecordId) {
+      openForm()
+    } else {
+      setError(true)
```

Comment:
Why are we setting this error here? If we want to make sure that recordid if formatted just add make `formatedRecordId` as the value of error below.